### PR TITLE
api:generate cli option

### PR DESCRIPTION
**Pull request by RomKadria**
add an option to integrate with cli package

### CODE REVIEW

File: src/commands/api/generate.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  async run() {
+    const setupApiProcess = spawn('npx', ['setup-api-monday'], { stdio: 'inherit' });
+
+    setupApiProcess.on('close', code => {
+      if (code === 0) {
+        logger.info('setup-api completed successfully.');
+      } else {
+        logger.error(`setup-api process exited with an error. Exit code: ${code}`);
+      }
+    });
```
Comment (DorShakedMonday) [inline src/commands/api/generate.ts:20]: Wrap in promise for better linear logic flow

File: src/commands/api/generate.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { spawn } from 'node:child_process';
+
+import { Command } from '@oclif/core';
+
+import logger from 'utils/logger';
+
+export default class ApiGenerate extends Command {
+  // TODO: yarn add @mondaycom/setup-api - when this package is published
+  static description = 'Run the setup-api-monday command from @mondaycom/setup-api';
```
Comment (DorShakedMonday) [inline src/commands/api/generate.ts:9]: align description to what is in the Readme and remove additions from the Readme (which were done manually)

File: src/commands/api/generate.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { spawn } from 'node:child_process';
+
+import { Command } from '@oclif/core';
+
+import logger from 'utils/logger';
+
+export default class ApiGenerate extends Command {
+  // TODO: yarn add @mondaycom/setup-api - when this package is published
```
Comment (DorShakedMonday) [inline src/commands/api/generate.ts:8]: remove when done please 🙏🏽

File: src/commands/api/generate.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  static description = 'Run the setup-api-monday command from @mondaycom/setup-api';
+
+  async run() {
+    const setupApiProcess = spawn('npx', ['setup-api-monday'], { stdio: 'inherit' });
+
+    setupApiProcess.on('close', code => {
+      if (code === 0) {
+        logger.info('setup-api completed successfully.');
+      } else {
+        logger.error(`setup-api process exited with an error. Exit code: ${code}`);
```
Comment (DorShakedMonday) [inline src/commands/api/generate.ts:18]: This error is to technical and should be client facing. Add with a `logger.debug` the exist code etc.. and in `logger.error` something more expressive for users

File: package.json [hunk 5]
```diff
@@ -7,10 +7,10 @@
   "bin": {
```
Comment (DorShakedMonday) [inline package.json:7]: Bump package version

Comment (DorShakedMonday) [reply package.json:7]: Run the `prepublish` command before commiting for auto generating some of the required files

**Review by DorShakedMonday — APPROVED** [review]

File: src/commands/api/generate.ts (hunk 1 above)
Comment (RomKadria) [reply src/commands/api/generate.ts:20]: done

File: src/commands/api/generate.ts (hunk 3 above)
Comment (RomKadria) [reply src/commands/api/generate.ts:8]: done

File: src/commands/api/generate.ts (hunk 4 above)
Comment (RomKadria) [reply src/commands/api/generate.ts:18]: done

File: package.json (hunk 5 above)
Comment (RomKadria) [reply package.json:7]: added to do

File: src/commands/api/generate.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                        run it from your root directory and it will create all neccesary files and scripts
+                        to start working with custom api queries and mutations.`;
+
+  async run() {
+    try {
+      await this.runSetupApi();
+      logger.info('setup-api completed successfully.');
+    } catch (error) {
+      logger.debug(error, this.DEBUG_TAG);
+      logger.error(`setup-api failed, please make sure you run it in your root directory and try again}`);
```
Comment (DorShakedMonday) [inline src/commands/api/generate.ts:20]: throw the error OR your error in the end so you would get the correct process.exit code

Comment (DorShakedMonday) [reply src/commands/api/generate.ts:20]: handled by the base classes

**Review by DorShakedMonday — APPROVED** [review]

File: src/commands/api/generate.ts (hunk 6 above)
Comment (RomKadria) [reply src/commands/api/generate.ts:19]: done

**Pull request merged**